src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3094,12 +3094,21 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
 
             return;
 
-            SynthesizedRecordConstructor addCtor(RecordDeclarationSyntax declWithParameters)
+            SynthesizedRecordConstructor? addCtor(RecordDeclarationSyntax declWithParameters)
             {
                 Debug.Assert(declWithParameters.ParameterList is object);
                 var ctor = new SynthesizedRecordConstructor(this, declWithParameters, diagnostics);
-                members.Add(ctor);
-                return ctor;
+
+                if (ctor.ParameterCount == 1 && ctor.Parameters[0].Type.Equals(this))
+                {
+                    diagnostics.Add(ErrorCode.ERR_RecordAmbigCtor, ctor.Locations[0]);
```

Comment:
>diagnostics.Add(ErrorCode.ERR_RecordAmbigCtor, ctor.Locations[0]); [](start = 20, length = 66)

I am not sure if we should perform the check and report this error for a case when a copy constructor is declared explicitly, we already report an error in that scenario.